jmespath-core/src/main/java/io/burt/jmespath/function/JoinFunction.java

```diff
@@ -0,0 +1,57 @@
+package io.burt.jmespath.function;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import io.burt.jmespath.Adapter;
+
+public class JoinFunction extends JmesPathFunction {
+  public JoinFunction() {
+    super(2, 2);
+  }
+
+  @Override
+  protected <T> T internalCall(Adapter<T> adapter, List<ExpressionOrValue<T>> arguments) {
+    T glue = arguments.get(0).value();
+    T components = arguments.get(1).value();
+    if (isStringArray(adapter, components)) {
+      if (adapter.isString(glue)) {
+        List<T> values = adapter.toList(components);
+        if (values.isEmpty()) {
+          return adapter.createString("");
+        } else {
+          StringBuffer buffer = new StringBuffer();
```

Comment:
I had used `StringBuffer` in half the cases and `StringBuilder` in half, it turns out. I fixed the other cases too.